lib/Sema/DerivedConformanceCodingKey.cpp

```diff
@@ -405,8 +405,9 @@ static bool canSynthesizeCodingKey(TypeChecker &tc, Decl *parentDecl,
       return false;
   }
 
-  if (!enumDecl->getInherited().empty() &&
-      enumDecl->getInherited().front().isError())
+  auto inherited = enumDecl->getInherited();
+  if (!inherited.empty() && inherited.front().wasValidated() &&
```

Comment:
Sure — I just wanted to introduce a small a change as possible, since checking `isError()` asserts if the type isn't validated. To be honest, is there even a reason to do this check or validation here? If the type doesn't validate, won't this just happen later anyway?